conan/cli/commands/search.py

```diff
@@ -1,24 +1,29 @@
 from collections import OrderedDict
 
 from conan.api.conan_api import ConanAPI
-from conan.cli.command import conan_command
-from conan.cli.commands.list import print_list_recipes, default_json_formatter
-
-
-# FIXME: "conan search" == "conan list recipes -r="*" -c" --> implement @conan_alias_command??
+from conan.cli.command import conan_command, OnceArgument
+from conan.cli.commands import default_json_formatter
+from conan.cli.commands.list import print_list_results
+from conan.internal.api.select_pattern import ListPattern
 from conans.errors import ConanException
 
 
-@conan_command(group="Consumer", formatters={"text": print_list_recipes, "json": default_json_formatter})
+# FIXME: "conan search" == "conan list recipes -r="*" -c" --> implement @conan_alias_command??
+@conan_command(group="Consumer", formatters={"text": print_list_results,
+                                             "json": default_json_formatter})
 def search(conan_api: ConanAPI, parser, *args):
     """
     Searches for package recipes in a remote or remotes
     """
-    parser.add_argument("query",
-                        help="Search query to find package recipe reference, e.g., 'boost', 'lib*'")
+    parser.add_argument('reference', help="Recipe reference or package reference. "
+                                          "Both can contain * as wildcard at any reference field. "
+                                          "If revision is not specified, it is assumed latest one.")
```

Comment:
```suggestion
    parser.add_argument('reference', help="Recipe reference to search for."
                                          "It can contain * as wildcard at any reference field.")
```
I think this is confusing. ``conan search`` is intended exclusively to search for recipe references, not for revisions, not for packages, etc. Everything else is a ``conan list`` command only.